File: leetcode/easy/4006-count-valid-prefixes/solution.cpp
class Solution {
public:
    int countValidPrefixes(string s) {
        int count=0;
        int n=s.size();
        int zero=0;
        int one=0;
        for(int i=0;i<n;i++){
            if(s[i]=='1'){
                one++;
            }
            else{
                zero++;
            }
            if(abs(zero-one)<=1){
            count++;
        }
    }
    return count;
    }
};

File: leetcode/easy/4006-count-valid-prefixes/solution.java
class Solution {
    public int countValidPrefixes(String s) {
        int count0=0;
        int count1=0;
        int ans=0;
        for (int i=0;i< s.length();i++){
            if (s.charAt(i) == '0') {
                count0++;
            }
            else{
                count1++;
            }
            if(Math.abs(count0 - count1)<=1){
            ans++;
        }
    } 
    return ans; 
    }
}